src/SignalR/common/Http.Connections/src/ConnectionEndpointRouteBuilderExtensions.cs

```diff
@@ -0,0 +1,144 @@
+        public static IEndpointConventionBuilder MapConnectionHandler<TConnectionHandler>(this IEndpointRouteBuilder builder, string pattern) where TConnectionHandler : ConnectionHandler
+        {
+            return builder.MapConnectionHandler<TConnectionHandler>(pattern, configureOptions: null);
+        }
+
+        /// <summary>
+        /// Maps incoming requests with the specified path to the provided connection pipeline.
+        /// </summary>
+        /// <typeparam name="TConnectionHandler">The <see cref="ConnectionHandler"/> type.</typeparam>
+        /// <param name="builder">The <see cref="IEndpointRouteBuilder"/></param>
+        /// <param name="pattern">The request path.</param>
+        /// <param name="configureOptions">A callback to configure dispatcher options.</param>
+        public static IEndpointConventionBuilder MapConnectionHandler<TConnectionHandler>(this IEndpointRouteBuilder builder, string pattern, Action<HttpConnectionDispatcherOptions> configureOptions) where TConnectionHandler : ConnectionHandler
+        {
+            var options = new HttpConnectionDispatcherOptions();
+            // REVIEW: WE should consider removing this and instead just relying on the
+            // AuthorizationMiddleware
+            var attributes = typeof(TConnectionHandler).GetCustomAttributes(inherit: true);
+            foreach (var attribute in attributes.OfType<AuthorizeAttribute>())
+            {
+                options.AuthorizationData.Add(attribute);
+            }
+            configureOptions?.Invoke(options);
+
+            var conventionBuilder = builder.MapConnections(pattern, options, b =>
+            {
+                b.UseConnectionHandler<TConnectionHandler>();
+            });
+
+            conventionBuilder.Add(e =>
+            {
+                // Add all attributes on the ConnectionHandler has metadata (this will allow for things like)
+                // auth attributes and cors attributes to work seamlessly
+                foreach (var item in attributes)
+                {
+                    e.Metadata.Add(item);
+                }
+            });
+
+            return conventionBuilder;
+        }
+
+
+        /// <summary>
+        /// Maps incoming requests with the specified path to the provided connection pipeline.
+        /// </summary>
+        /// <param name="builder">The <see cref="IEndpointRouteBuilder"/></param>
+        /// <param name="pattern">The request path.</param>
+        /// <param name="options">Options used to configure the connection.</param>
+        /// <param name="configure">A callback to configure the connection.</param>
+        public static IEndpointConventionBuilder MapConnections(this IEndpointRouteBuilder builder, string pattern, HttpConnectionDispatcherOptions options, Action<IConnectionBuilder> configure)
+        {
+            var dispatcher = builder.ServiceProvider.GetRequiredService<HttpConnectionDispatcher>();
```

Comment:
What registers `HttpConnectionDispatcher`? If it isn't in the box by default then add a check for it (or another marker service) and throw a friendly error message like happens in `MapHub`.

Reply:
Yes it could, this is one of those low low level APIs though. Not many people (if any) use it.